Docker solver

pavolbauer opened the pull request.

Aratz committed:
Add engine class documentation

Aratz committed:
Add sample_candidate documentation

Aratz committed:
Add run_simulation documentation

Aratz committed:
Add kolmogorov_distance documentation

Aratz committed:
Add run documentation

Aratz committed:
Add kolmogorov_distance unit tests

Aratz committed:
Clarify documentation

It was a little unclear what the input data should be.

Aratz committed:
Implement kolmogorov_distance

Aratz committed:
Add precondition to kolmogorov_distance

Aratz committed:
Add engine constructor unit tests

Aratz committed:
Add engine constructor

Aratz committed:
Merge branch 'master' into dev/constructor

Aratz committed:
Autoformat code

Aratz committed:
Reorganize imports

Aratz committed:
Mark missing functions with NotImplementedError

Aratz committed:
Restrict priors to uniform distributions

To facilitate particle sampling, the program now only supports uniform
prior distributions, the hashmap associates parameter names to the
bounds of their uniform distribution.

Aratz committed:
Add sampling unit tests

Aratz committed:
Switch parameter fields to public

Aratz committed:
Add sampling_candidate

pavolbauer committed:
initial commit modified INSIGHT code + docker file

Commit:
attempting scala function

pavolbauer committed:
working function+test

Aratz commented:
There are a few bugs in the code that make the build fail but I need to add stuff anyways so I'll try to fix it, @pavolbauer I'll come to you if I need help with your code. I schedule to work on this tomorrow.

Commit:
fixed scala bug

pavolbauer committed:
serial solver takes list of particles now

pavolbauer committed:
Merge branch 'docker_solver' of https://github.com/mcapuccini/cloud-insight into docker_solver

Commit:
modified pom and travis file

Commit:
particle evaluate takes and returns list now

Commit:
from easymapreduce example

Commit:
fixing with marco

Commit:
fixed unit test

pavolbauer committed:
serial solver now reads JSON vectors

pavolbauer committed:
dockerfile using debian

pavolbauer committed:
travis file updated

pavolbauer committed:
passing tests locally

pavolbauer committed:
formatted code

Commit:
Update README.md

Aratz committed:
Switch to Lists to represent parameters

It turned out it was a bit tricky to convert a Hashmap into a vector and
keep the order of the parameters correct. The new implementation is now
a bit more bulky and it is now up to the user to put the parameter's
values in the right order, i.e. the order that should be used for the
solver.

Aratz committed:
Fix particle sampling test

Aratz committed:
Add run function and test on weights

Aratz committed:
Format code

Aratz commented:
I think it should be good to merge now.
